### DISCUSSION TITLE
the program takes a long time to produce results

### DISCUSSION THREAD

**Discussion by guvensimsek** [kogalur/randomForestSRC#125] [Q&A] [answered]
Hi,
I have the survival data set (with 1954072 rows 11 columns). When I run the program with rfsrc function, it takes a very long time. Is there an easy way to shorten this time? 

Thank you in advance for your interest.

**Comment by ishwaran** [comment, marked as answer]
Yes, there are some easy things you can do.

First you should set nodesize to a reasonably large value.  For large n, there shouldn't be appreciable loss of performance.  So try `nodesize=1000`.

However, there is a much bigger problem at play, which is the C-index calculation.  This is order O(n^2) so with your sample size we are talking ~ 4x10^12 calculations!

The solution is to set the hidden option perf.type to none.  So `perf.type="none"`.  This will turn off all performance (so you won't get an OOB error rate anymore).

The generic survival call for large n is something like:

`o <- rfsrc(Surv(my.time,my.cens)~., my.data, ntree=50, nodesize=1000, perf.type="none")`

I tried the above with n=2 million cases and p = 20 and it compiles in about 248 seconds.

Also see the following vignette:

https://luminwin.github.io/randomForestSRC/articles/speedup.html

**Reply by guvensimsek** [reply]
Thanks for your response. Also, I congratulate you for your contributions in this field.
